carbon-design-system/carbon-addons-iot-react
# feat(page-title-bar): design audit and add new stackBreadcrumbsWithTabs feature

Closes #3045 

**Summary**

- There were a few minor design issues that needed to be addressed. These were minimal and simple, but the big new addition is the design requirements for sticky navigation has changed. The new behavior causes both the breadcrumbs and the page tabs to be sticky in dynamic and condensed modes. This behavior is added behind the `stackBreadcrumbsWithTabs` prop. If this prop is true (it defaults to false to maintain current behavior), when the header is in dynamic or condensed modes, if you continue to scroll passed the tabs it will pick them up and make them sticky, too. If the `stackBreadcrumbsWithTabs` is false, it will cause the breadcrumbs to get pushed off the screen, but new behavior was added to force them to re-appear when you scroll back up.

**Change List (commits, features, bugs, etc)**

- add 1px space between buttons in the header-right area
- add padding on tab edge and bottom border to separate tabs from page content
- add `stackBreadcrumbsWithTabs` knob to all stories
- add style fixes for stacked tabs/breadcrumbs
- add behavior to stack sticky breadcrumbs and tabs when in dynamic or condensed modes
- add behavior to force the breadcrumbs to re-appear when you scroll up (quickly--more than 20px) when `stackBreadcrumbsWithTabs` is false
- update all snapshots
- add tests to cover all new behavior

**Acceptance Test (how to verify the PR)**

- on the https://deploy-preview-3080--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-pagetitlebar--with-dynamic-scrolling-and-tabs story
- scroll down and confirm breadcrumbs condense, and stay sticky for a few pixels, but are then replaced by the tabs being sticky
- scroll back up quickly
- confirm breadcrumbs appear and both breadcrumbs and tabs are both sticky and stack together
- scroll back down and confirm breadcrumbs hide again
- turn on `stackBreadcrumbsWithTabs` knob
- confirm tabs get picked up and remain sticky below the breadcrumbs always--even when scrolling up or back down
- change to condensed mode
- confirm tabs and breadcrumbs are always stacked and sticky together when scrolling down
- turn off `stackBreadcrumbsWithTabs` knob 
- confirm tabs are never sticky in condensed mode

**Regression Test (how to make sure this PR doesn't break old functionality)**

- compare other PageTitleBar stories against next to ensure no other behavior broken



**Things to look for during review**
